File: Signal.CPP/IntroSignal.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "IntroSignal.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TIntroSignalForm *IntroSignalForm;
//---------------------------------------------------------------------------
__fastcall TIntroSignalForm::TIntroSignalForm(TComponent* Owner)
	: TBasicForm3(Owner)
{
	RichEdit1->Lines->Clear();
	RichEdit1->DefAttributes->Size = 12;
	RichEdit1->DefAttributes->Style = RichEdit1->DefAttributes->Style << TFontStyle::fsBold;
	RichEdit1->SelAttributes->Color = (TColor) RGB(51,51,153);
	RichEdit1->SelAttributes->Size = 14;
	RichEdit1->Lines->Add("A wide range of signal filtering routines includes:");
	RichEdit1->Lines->Add("");
	RichEdit1->Paragraph->Numbering = nsBullet;
	RichEdit1->Paragraph->FirstIndent = 10;
	RichEdit1->Lines->Add("Elliptic,Butterworth and Chebyshev filter designers");
	RichEdit1->Lines->Add("Parks-McClellan optimal FIR filter design routine");
	RichEdit1->Lines->Add("Savitzky-Golay polynomial FIR filter");
	RichEdit1->Lines->Add("Multi-rate, multi-stage decimation and interpolation filters");
	RichEdit1->Lines->Add("Envelope detection/amplitude demodulation");
	RichEdit1->Lines->Add("Zoom-spectrum filter");
	RichEdit1->Paragraph->Numbering = nsNone;
	RichEdit1->Paragraph->FirstIndent = 0;
}
//---------------------------------------------------------------------------

File: Signal.Delphi/IntroSignal.pas
unit IntroSignal;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic3, StdCtrls, ComCtrls;

type
  TIntroSignalForm = class(TBasicForm3)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  IntroSignalForm: TIntroSignalForm;

implementation

{$IFDEF CLR}
{$R *.nfm}
{$ELSE}
{$R *.dfm}
{$ENDIF}

procedure TIntroSignalForm.FormCreate(Sender: TObject);
begin
  inherited;
  With RichEdit1.Lines, RichEdit1 do
  begin
    Clear;
    DefAttributes.Size := 10;
    DefAttributes.Style := [fsBold];
    SelAttributes.Size := 12;

    SelAttributes.Color := RGB(51,51,153);
    Add('A wide range of signal filtering routines includes:');
    Add('');
    Paragraph.FirstIndent := 10;
    Paragraph.Numbering := nsBullet;
    Add('Elliptic,Butterworth, Bessel and Chebyshev filter designers');
    Add('Parks-McClellan optimal FIR filter design routine');
    Add('Savitzky-Golay polynomial FIR filter');
    Add('Windowed FIR filter design with Kaiser window.');
    Add('Simple DC filters, Median filters, Moving average, Exponential decay and delay filters.');
    Add('Fractional delay filters.');
    Add('Signal generators.');
    Paragraph.Numbering := nsNone;    
    Paragraph.FirstIndent := 0;
  end;
end;

initialization
  RegisterClass(TIntroSignalForm);

end.
